### ISSUE TITLE
Hedera Testnet Reset: Further Testing & Document RPC Relay behaviors & MetaMask

### ISSUE THREAD

**Issue by a-ridley** [hashgraph/hedera-json-rpc-relay#1580]
### Background Information
On July 27th, there was a testnet reset. 
On that same day (before the reset), I was working with a testnet account created through the Hedera portal (_created before the reset_). I imported the account into MetaMask(_imported account before the reset occurred_) and added the Hedera testnet to MetaMask using the following RPC URL: `https://testnet.hashio.io/api`. I used [Remix](https://remix.ethereum.org/) to deploy a smart contract using my imported hedera account in MetaMask, and it deployed successfully. (before the reset occurred).
 
 This portal account went through the reset, (I had the portal account website opened and noticed the new account id). I continued working with my account on MetaMask and continued to execute contract calls through Remix, and MetaMask was no longer behaving as expected .
 
 Some of the observed issues:
 * Remix constantly showed that it could not "connect to the network." 
 * Switching networks on MetaMask and back to Hedera Testnet would fail to connect.  
 * MetaMask continued to show old account transaction data that occurred before the reset
 * MetaMask displayed 0 HBAR for the portal account
 * Relay would return different errors. Some would say.`Response has no error or result for request`,` nonce to low`
 
 Eventually, I removed the MetaMask extension completely. Re-installed the extension, re-added the Hedera network, & re-imported my account and everything behaved as expected.
 
 I suspect MetaMask gets into a weird state when a Testnet reset is done and it could be that the communication between the Relay and MetMask gets tricky. The first reset took place back in January, and MetaMask support was still in the works. The July 27th reset is now the second testnet reset, & MetaMask support has been added. It is possible there is uncharted workflows with possible bugs that are caused during a testnet reset that is affecting how the RPC Relay and MetaMask communicate that should be tested and documented.

### Steps to reproduce

1. Set up and run a local testnet instance
2. Have the MetaMask extension v.10.34.1
3. Add the Hedera testnet network with the following RPC URL: `https://testnet.hashio.io/api`, Chain ID: `296`, Symbol: `HBAR`
4. Import your portal account into MetaMask
5. Open [Remix](https://remix.ethereum.org/)
6. In the File Explorer add a new file called `SampleContract.sol` in the contracts directory & create the sample contract with the following:
```
// SPDX-License-Identifier: GPL-3.0
pragma solidity >=0.7.0 <0.8.9;

import "https://github.com/hashgraph/hedera-smart-contracts/blob/v0.2.0/contracts/hts-precompile/HederaResponseCodes.sol";
import "https://github.com/hashgraph/hedera-smart-contracts/blob/v0.2.0/contracts/hts-precompile/IHederaTokenService.sol";
import "https://github.com/hashgraph/hedera-smart-contracts/blob/v0.2.0/contracts/hts-precompile/HederaTokenService.sol";
import "https://github.com/hashgraph/hedera-smart-contracts/blob/v0.2.0/contracts/hts-precompile/ExpiryHelper.sol";

contract SampleContract is ExpiryHelper {
  event CreatedToken(address tokenAddress);
  event Response(int response);

  address public fungibleAddress;
  address public owner;

  constructor() payable {
    IHederaTokenService.HederaToken memory token;
    token.name = "Sample Tokens";
    token.symbol = "ST";
    token.memo = "ST Tokens Rule";
    token.treasury = address(this);
    token.expiry = createAutoRenewExpiry(address(this), 7000000);

    (int responseCode, address tokenAddress) = HederaTokenService
      .createFungibleToken(token, 1000, 0);

    if (responseCode != HederaResponseCodes.SUCCESS) {
      revert();
    }

    fungibleAddress = tokenAddress;
    owner = msg.sender;
    emit CreatedToken(tokenAddress);
  }

  function createNFT(string memory name, string memory symbol) external payable {
    IHederaTokenService.TokenKey[]
      memory keys = new IHederaTokenService.TokenKey[](1);

    // Set this contract as supply
    keys[0] = getSingleKey(
      KeyType.SUPPLY,
      KeyValueType.CONTRACT_ID,
      address(this)
    );

    IHederaTokenService.HederaToken memory token;
    token.name = name;
    token.symbol = symbol;
    token.memo = "Sample Collection";
    token.treasury = address(this);
    token.tokenSupplyType = true;
    token.maxSupply = 10;
    token.tokenKeys = keys;
    token.freezeDefault = false;
    token.expiry = createAutoRenewExpiry(address(this), 7000000);

    (int responseCode, address createdToken) = HederaTokenService
      .createNonFungibleToken(token);

    if (responseCode != HederaResponseCodes.SUCCESS) {
      revert("Failed to create non-fungible token");
    }

    emit CreatedToken(createdToken);
  }
 }
```
7. Go to the Deploy & Run Transactions tab & change the Environment section to connect to your Hedera account that is imported in MetaMask (Injected Provider - MetaMask) & ensure the Account section is indeed your Hedera account EVM address.
8. Set the Value section to 20 `Ether`
9. Deploy your contract, create an nft
10. Complete or simulate a Testnet Reset on your local testnet instance 
11. After the reset is complete, attempt to complete `steps 7 - 9`
12. Observe the how the JSON RPC Relay behaves & how MetaMask behaves 
13. Capture/document

### Hedera network

testnet

### Version

v0.28.0

### Operating system

macOS

**Comment by Nana-EC** [comment]
Thanks @a-ridley .
Just to clarify for other readers Metamask had support for before the Jan reset however, it was likely less usage so not many encountered the unique effects of a testnet reset.

I think we likely need to document that users should expect to reset their wallet account after a reset.

In the portal case the balance indeed would have changed and all previous transactions would not be applicable in the test environment.
Essentially it's a new account with the same private key and hence EVM address.
Non portal cases will be different as the Hedera account would be non existent. A transfer to the EVM address would be needed to create a hollow account with a matching address.

As part of this there's likely some Metamask caching and confusion we'll have to confirm but the wallet account reset should capture most of the work.

We'll take a look.
Thanks for the details.

**Comment by kabaluyot** [comment]
Experienced this too:

- all accounts reverted to 0 balance
- cannot send HBAR to another account anymore using MetaMask (same with ERC20 tokens)

Currently using own testnet hedera rpc node

**Comment by Nana-EC** [comment]
@kabaluyot you'll have to recreate those accounts as you did before.
If you used the Hedera portal they should have been recreated.
To maintain your exact account just send some HBAR to the appropriate evm address to cause an autocreate and wallet fund.

Testnet will be reset every 3 months, see https://docs.hedera.com/hedera/networks/testnet#test-network-resets

**Comment by kabaluyot** [comment]
Yes was able to recreate the accounts. But sending HBAR/ERC20 tokens right now in MetaMask is hit and miss.

**Comment by Ferparishuertas** [comment]
Close due inactivity. Feel free to reopen if still happening
